ghcup-cross-0.0.8.yaml

```diff
@@ -27,30 +27,29 @@ ghcupDownloads:
               dlUri: https://downloads.haskell.org/~ghcup/unofficial-bindists/ghc/javascript-unknown-ghcjs-9.6.2/ghc-javascript-unknown-ghcjs-9.6.2-aarch64-apple-darwin.tar.xz
               dlHash: 0d65bb3940a820d95b4b33ea0cd61bf39670d48c4d8f747153ded01a7ef34821
               dlSubdir: ghc-9.6.2-javascript-unknown-ghcjs
-    wasm32-wasi-9.6.3.20230927:
+    wasm32-wasi-9.6.4:
```

Comment:
Why replace and not extend?

Reply:
The old bindists must be used with an old `wasi-sdk`, since our recent update in the toolchain contains ABI breaking change. Given the official installation guide still uses `ghc-wasm-meta`'s `master` branch which will contain the new toolchain, it makes sense to simply replace the old bindists.

Reply:
The idea generally in ghcup is to be append only.

We can print a post-install message to tell users to use a specific `ghc-wasm-meta` commit to use with the bindist version.

Reply:
What is the ghc-wasm-meta commit that will work?

Reply:
ping

Reply:
Are you able to add a note to the bindist that need it?

E.g.:

```yaml
      viPostInstall: &old-wasm-post-install |
        To use this bindist, you need to use the commit c0aa3bb7d88bb6ec809210e17658dd1ed64ba66c of ghc-wasm-meta repository.
        Also see: https://www.haskell.org/ghcup/guide/#ghc-wasm-cross-bindists-experimental
```

And then you can re-use that yaml anchor per bindist such as:

```yaml
      viPostInstall: *old-wasm-post-install
```

And then ideally send a PR to also update ghcup documentation: https://github.com/haskell/ghcup-hs/blob/master/docs/guide.md#ghc-wasm-cross-bindists-experimental

Reply:
til `viPostInstall`, thanks and i'll push the change later. where is this metadata documented, may i ask?

Reply:
There's no comprehensive documentation about it.

You basically have to follow the types: https://github.com/haskell/ghcup-hs/blob/190d308ddf4a95f2024f9fc3ca8a1f14d3f62711/lib/GHCup/Types.hs#L76-L81

We tried to do a json schema once, but that stalled.

Reply:
I just noticed that post-install is not really appropriate, because you need the wasm environment sourced before installing the cross GHC. We need pre install message. I'll have to add that as a new feature for the next release.

Reply:
https://github.com/haskell/ghcup-hs/issues/1016

Reply:
docs change: https://github.com/haskell/ghcup-hs/pull/1017